prototypes/ItemRepeaterPrototypes/ItemsRepeaterExperiments/AttachedBehaviors/SelectionBehavior.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Microsoft.UI.Xaml.Controls;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Controls;
+
+namespace ItemsRepeaterExperiments.AttachedBehaviors
+{
+    public class SelectionBehavior : DependencyObject
+    {
+        // Should we store this as an attached property on the repeater?
+        // What would happen with to repeaters and shift click?
+        private static UIElement LastFocused;
+
+        public static bool GetIsEnabled(DependencyObject obj)
+        {
+            return (bool)obj.GetValue(IsEnabledProperty);
+        }
+
+        public static void SetIsEnabled(DependencyObject obj, bool value)
+        {
+            obj.SetValue(IsEnabledProperty, value);
+        }
+
+        public static readonly DependencyProperty IsEnabledProperty =
+            DependencyProperty.RegisterAttached("IsEnabled", typeof(bool), typeof(SelectionBehavior), new PropertyMetadata(false,
+                new PropertyChangedCallback(OnIsEnabledChanged)));
+
+        private static void OnIsEnabledChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+        {
+            if((bool)args.NewValue)
+            {
+                SetIsSelected(sender, GetIsEnabled(sender));
+                (sender as UIElement).PointerPressed += SelectionBehavior_PointerPressed;
+            }
+            else
+            {
+                (sender as UIElement).PointerPressed -= SelectionBehavior_PointerPressed;
+            }
+        }
+
+
+        private static void SelectionBehavior_PointerPressed(object sender, Windows.UI.Xaml.Input.PointerRoutedEventArgs e)
+        {
+            var repeater = GetParentItemsRepeater(sender as DependencyObject);
+            var selectionModel = GetSelectionModel(sender as DependencyObject);
+            var elementIndex = repeater.GetElementIndex(sender as UIElement);
+            if (selectionModel.SingleSelect)
+            {
+                if (e.KeyModifiers == Windows.System.VirtualKeyModifiers.Control)
+                {
+                    selectionModel.Deselect(elementIndex);
+                }
+                else
+                {
+                    selectionModel.Select(elementIndex);
+                }
+            }
+            else
+            {
+                switch (e.KeyModifiers)
+                {
+                    case Windows.System.VirtualKeyModifiers.Shift:
+                        if (PointerBehaviors.LastFocused != null)
```

Comment:
Is there a way to avoid tracking the LastFocused ? That seems like it can cause some trouble. What if we moved the anchor as items get focus right away ? If range selection is not done then it does not get used and it will be a no-op

Reply:
Yeah that wasn't the best idea. I've switched it to set the anchor to the last item we (de)selected.